cli/src/main/java/com/devonfw/tools/ide/repo/MavenRepository.java

```diff
@@ -0,0 +1,158 @@
+package com.devonfw.tools.ide.repo;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.os.SystemInfo;
+import com.devonfw.tools.ide.url.model.file.UrlDownloadFileMetadata;
+import com.devonfw.tools.ide.url.model.file.json.ToolDependency;
+import com.devonfw.tools.ide.version.GenericVersionRange;
+import com.devonfw.tools.ide.version.IdeVersion;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+
+/**
+ * Implementation of {@link AbstractToolRepository} for maven-based artifacts.
+ */
+public class MavenRepository extends AbstractToolRepository {
+
+  private static final String MAVEN_CENTRAL = "https://repo1.maven.org/maven2";
+
+  private static final String MAVEN_SNAPSHOTS = "https://s01.oss.sonatype.org/content/repositories/snapshots";
+
+  private static final String MAVEN_METADATA_XML = "maven-metadata.xml";
+
+  /** Group id of IDEasy. */
+  public static final String IDEASY_GROUP_ID = "com.devonfw.tools.IDEasy";
+
+  /** Artifact Id of IDEasy. */
+  public static final String IDEASY_ARTIFACT_ID = "ide-cli";
+
+  /**
+   * The constructor.
+   *
+   * @param context the owning {@link IdeContext}.
+   */
+  public MavenRepository(IdeContext context) {
+
+    super(context);
+  }
+
+  private String getPath(String groupId, String artifactId, String version, String fileName) {
+
+    String basePath = groupId.replace('.', '/') + "/" + artifactId;
+    return version != null ? basePath + "/" + version + "/" + fileName : basePath + "/" + fileName;
+  }
+
+  private String getBaseUrl(String groupId, String artifactId) {
+
+    if (isIdeasySnapshot(groupId, artifactId, IdeVersion.get())) {
+      return MAVEN_SNAPSHOTS;
+    }
+    return MAVEN_CENTRAL;
+  }
```

Comment:
this and its usage is wrong.
You implemented `MavenRepository` specific for our use-case of downloading IDEasy.
`IdeVersion.get()` needs to be used outside of this class to do the version comparison.
In `getMetadata` you need to use the `version` argument that you ignoring here.